int32 data
int32 x
int32 y
int32 z
uint64 ROStimestamp
int64 MCtimestamp